app/models/concerns/orchestration.rb

```diff
@@ -47,24 +47,18 @@ def rollback
     raise ActiveRecord::Rollback
   end
 
-  # log and add to errors
-  def failure(msg, exception_or_backtrace = nil, dest = :base)
-    if exception_or_backtrace
-      if exception_or_backtrace.is_a? Exception
-        exception = exception_or_backtrace
-      else
-        exception = StandardError.new(msg)
-        exception.set_backtrace(exception_or_backtrace)
-        Foreman::Deprecation.deprecation_warning("1.11", "Passing backtrace to failure method is deprecated, pass the exception instead")
-      end
-      Foreman::Logging.exception(msg, exception)
-    else
-      logger.warn(msg)
-    end
-    errors.add dest, msg
+  # Log and add error to model
+  def failure(message, exception = nil, dest = :base)
```

Comment:
This method was "semi-deprecated" - passing backtraces as the 2nd argument was deprecated, so now it handles only logging exceptions and regular messages.